src/pages/chat/chat.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+import TopBar from "../../components/TopBar/topbar";
+import ChatInput from "../../components/ChatInput/chatinput";
+import userData from "../../assets/datas/userdata.json";
+import chatData from "../../assets/datas/chatdata.json";
+import ChatTitle from "../../components/ChatTitle/chatTitle";
+import MessageList from "../../components/Message/messageList";
+import Message from "../../components/Message/message";
+import "../../styles/colors.css";
+import bottomBar from "../../assets/images/LightBottomBar.svg";
+import StatusBar from "../../components/StatusBar/statusbar";
+
+interface Message {
+  id: number;
+  sender: string;
+  content: string;
+  showIcon: boolean;
+}
+
+function Chat() {
+  const savedChats = JSON.parse(
+    localStorage.getItem("chatMessages") || JSON.stringify(chatData)
+  );
+  const savedMessageId = JSON.parse(
+    localStorage.getItem("currMessageId") || JSON.stringify(chatData.length)
+  );
+  const [messages, setMessages] = useState<Message[]>(savedChats);
+  const [messageId, setMessageId] = useState<number>(savedMessageId);
+  const [nowUser, setNowUser] = useState(userData.users[0]);
+
+  const updateMessagesWithShowIcon = (messages: Message[]) => {
+    const updatedMessages = [...messages];
+
+    for (let i = 1; i < updatedMessages.length; i++) {
```

Comment:
```suggestion
   updatedMessages.forEach((message, i, arr) => {
  if (i > 0 && message.sender !== arr[i - 1].sender) {
    message.showIcon = true;
  }
});

```
이렇게 forEach 매서드를 사용하면 가독성이 조금 더 좋아질 것 같아요!